WalletWasabi.Tests/UnitTests/WabiSabi/Crypto/CredentialRequesting/ZeroCredentialsRequestTests.cs

```diff
@@ -0,0 +1,51 @@
+using NBitcoin.Secp256k1;
+using System.Linq;
+using WalletWasabi.Crypto;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Crypto.ZeroKnowledge;
+using WalletWasabi.WabiSabi.Crypto;
+using WalletWasabi.WabiSabi.Crypto.CredentialRequesting;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests.WabiSabi.Crypto.CredentialRequesting
+{
+	/// <summary>
+	/// Tests for <see cref="ZeroCredentialsRequest"/> class.
+	/// </summary>
+	public class ZeroCredentialsRequestTests
+	{
+		[Fact]
+		public void EqualityTest()
+		{
+			// Request #1.
+			ZeroCredentialsRequest request1 = GetZeroCredentialsRequest(1);
+
+			//// Request #2.
+			ZeroCredentialsRequest request2 = GetZeroCredentialsRequest(1);
+
+			Assert.Equal(request1, request2);
+
+			//// Request #3.
+			ZeroCredentialsRequest request3 = GetZeroCredentialsRequest(2);
+			Assert.NotEqual(request1, request3);
+		}
+
+		private static GroupElement NewGroupElement(int i) => Generators.FromText($"T{i}");
+
+		private static GroupElementVector NewGroupElementVector(params int[] arr) => new(arr.Select(i => NewGroupElement(i)));
+
+		private static ScalarVector NewScalarVector(params uint[] arr) => new(arr.Select(i => new Scalar(i)));
```

Comment:
I need to reread some of the old code but it seems like these can clean up some of the boilerplate I added in `ProofSystem` and the unit tests

Reply:
I agree that it would be great to have these helper methods defined somewhere for tests to use. Currently, it's being copied which is normally a bad thing but in these tests, it should be OK (at least for now).

Reply:
i don't see a problem with this being added here at all, i don't mind repetition/boilerplate in tests, i just had no idea `params` was a thing when i wrote that ;-)